src/atlas/grid/detail/grid/Gaussian.cc

```diff
@@ -246,6 +246,28 @@ StructuredGrid::grid_t* reduced_gaussian( const std::vector<int>& nx, const Doma
     return new StructuredGrid::grid_t( xspace( _nx ), Spacing( yspace ), Projection(), domain );
 }
 
+StructuredGrid::grid_t* reduced_gaussian( const std::vector<int> & nx, double centre[], double stretch) {
+  using namespace atlas::util;
+
+  std::vector<Spacing> spacings (nx.size ());
+
+  for (int i = 0; i < nx.size (); i++)
+    {   
+      double lonmax = 360.0 * double (nx[i] - 1) / double (nx[i]);
+      spacings[i] = Spacing (Config ("type", "linear") | Config ("N", nx[i])
+                           | Config ("start", 0) | Config ("end", lonmax));
+    }   
+
+  StructuredGrid::XSpace xspace (spacings);
```

Comment:
I believe you could xspace more simply:
```
StructuredGrid::XSpace xspace(nx);
```
It would create a linear spacing between 0. and 360, which excludes 360. itself.